pennylane/pulse/parametrized_hamiltonian_pytree.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing the ``JaxParametrizedHamiltonian`` class."""
+from dataclasses import dataclass
+from typing import Callable, Tuple, Union
+
+import pennylane as qml
+
+from .parametrized_hamiltonian import ParametrizedHamiltonian
+
+has_jax = True
+try:
+    import jax
+    import jax.numpy as jnp
+    from jax.experimental import sparse
+    from jax.tree_util import register_pytree_node_class
+except ImportError:
+    has_jax = False
+
+
+@register_pytree_node_class
+@dataclass
+class ParametrizedHamiltonianPytree:
+    """Jax pytree class that represents a ``ParametrizedHamiltonian``."""
+
+    mat_fixed: Union[jnp.ndarray, sparse.CSR]
+    mats_parametrized: Tuple[Union[jnp.ndarray, sparse.CSR], ...]
+    coeffs_parametrized: Tuple[Callable]
+
+    @staticmethod
+    def from_hamiltonian(H: ParametrizedHamiltonian, dense: bool = False, wire_order=None):
+        """Convert a ``ParametrizedHamiltonian`` into a jax pytree object.
+
+        Args:
+            H (ParametrizedHamiltonian): parametrized Hamiltonian to convert
+            dense (bool, optional): Decide wether a dense/sparse matrix is used. Defaults to False.
+            wire_order (list, optional): Wire order of the returned ``JaxParametrizedOperator``.
+                Defaults to None.
+
+        Returns:
+            ParametrizedHamiltonianPytree: pytree object
+        """
+        if not has_jax:
+            raise ImportError(
+                "Module jax is required for the ``ParametrizedHamiltonianPytree`` class. "
+                "You can install jax via: pip install jax"
+            )
+        make_array = jnp.array if dense else sparse.CSR.fromdense
+
+        if len(H.ops_fixed) > 0:
+            mat_fixed = make_array(qml.matrix(H.H_fixed(), wire_order=wire_order))
+        else:
+            mat_fixed = None
+
+        mats_parametrized = tuple(
+            make_array(qml.matrix(op, wire_order=wire_order)) for op in H.ops_parametrized
+        )
+        return ParametrizedHamiltonianPytree(mat_fixed, mats_parametrized, H.coeffs_parametrized)
+
+    def __call__(self, pars, t):
+        if self.mat_fixed is not None:
+            ops = (self.mat_fixed,)
+            coeffs = (1,)
+        else:
+            ops = ()
+            coeffs = ()
+        coeffs = coeffs + tuple(f(p, t) for f, p in zip(self.coeffs_parametrized, pars))
```

Comment:
cc @Qottmann is this correct? the `pars` are zipped together with the `coeffs_parametrized`?

Reply:
Yes that should work, its the same as what qml.dot does